### ISSUE TITLE
loss NaN during training decoder

### ISSUE THREAD

**Issue by jsrdcht** [tardisblue9/DeDe#4]
When I train the decoder model, I encounter an issue where the loss value becomes NaN. The infected encoder is trained by badencoder and every setting is default. Grad clipping is not helping.

```shell
Decoder Train Epoch: [65/200] Loss: 0.4426: 100%|███████████████████████████████████████████████████████| 195/195 [00:31<00:00,  6.12it/s]
Decoder Train Epoch: [66/200] Loss: 0.4419: 100%|███████████████████████████████████████████████████████| 195/195 [00:30<00:00,  6.45it/s]
Decoder Train Epoch: [67/200] Loss: 0.4383: 100%|███████████████████████████████████████████████████████| 195/195 [00:28<00:00,  6.90it/s]
Decoder Train Epoch: [68/200] Loss: 0.4358: 100%|███████████████████████████████████████████████████████| 195/195 [00:32<00:00,  5.99it/s]
Decoder Train Epoch: [69/200] Loss: 0.4351: 100%|███████████████████████████████████████████████████████| 195/195 [00:27<00:00,  7.11it/s]
Decoder Train Epoch: [70/200] Loss: 0.4324: 100%|███████████████████████████████████████████████████████| 195/195 [00:32<00:00,  5.91it/s]
Decoder Train Epoch: [71/200] Loss: 0.4338: 100%|███████████████████████████████████████████████████████| 195/195 [00:28<00:00,  6.95it/s]
Decoder Train Epoch: [72/200] Loss: 0.4316: 100%|███████████████████████████████████████████████████████| 195/195 [00:29<00:00,  6.55it/s]
Decoder Train Epoch: [73/200] Loss: 0.4311: 100%|███████████████████████████████████████████████████████| 195/195 [00:30<00:00,  6.36it/s]
Decoder Train Epoch: [74/200] Loss: 0.4273: 100%|███████████████████████████████████████████████████████| 195/195 [00:27<00:00,  7.10it/s]
Decoder Train Epoch: [75/200] Loss: 0.4280: 100%|███████████████████████████████████████████████████████| 195/195 [00:25<00:00,  7.72it/s]
Decoder Train Epoch: [76/200] Loss: 0.4268: 100%|███████████████████████████████████████████████████████| 195/195 [00:22<00:00,  8.63it/s]
Decoder Train Epoch: [77/200] Loss: 0.4268: 100%|███████████████████████████████████████████████████████| 195/195 [00:22<00:00,  8.85it/s]
Decoder Train Epoch: [78/200] Loss: 0.4221: 100%|███████████████████████████████████████████████████████| 195/195 [00:22<00:00,  8.81it/s]
Decoder Train Epoch: [79/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:22<00:00,  8.85it/s]
Decoder Train Epoch: [80/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:26<00:00,  7.26it/s]
Decoder Train Epoch: [81/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:31<00:00,  6.22it/s]
Decoder Train Epoch: [82/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:32<00:00,  6.07it/s]
Decoder Train Epoch: [83/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:28<00:00,  6.73it/s]
Decoder Train Epoch: [84/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:33<00:00,  5.86it/s]
Decoder Train Epoch: [85/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:28<00:00,  6.78it/s]
Decoder Train Epoch: [86/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:28<00:00,  6.73it/s]
Decoder Train Epoch: [87/200] Loss: nan: 100%|██████████████████████████████████████████████████████████| 195/195 [00:32<00:00,  5.94it/s]
```

**Comment by jsrdcht** [comment]
This error may be due to an incorrect weight decay setting. Parameters such as the bias of linear weights and positional bias in transformers do not require weight decay.

**Comment by jsrdcht** [comment]
This error may be due to an incorrect weight decay setting. Parameters such as the bias of linear weights and positional bias in transformers do not require weight decay.
